File: src/main/java/com/skyspaceeleven/service/ServiceService.java
package com.skyspaceeleven.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeleven.domain.Service;
import com.skyspaceeleven.dto.ServiceDTO;
import com.skyspaceeleven.dto.ServiceSearchDTO;
import com.skyspaceeleven.dto.ServicePageDTO;
import com.skyspaceeleven.dto.ServiceConvertCriteriaDTO;
import com.skyspaceeleven.service.GenericService;
import com.skyspaceeleven.dto.common.RequestDTO;
import com.skyspaceeleven.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ServiceService extends GenericService<Service, Integer> {

	List<Service> findAll();

	ResultDTO addService(ServiceDTO serviceDTO, RequestDTO requestDTO);

	ResultDTO updateService(ServiceDTO serviceDTO, RequestDTO requestDTO);

    Page<Service> getAllServices(Pageable pageable);

    Page<Service> getAllServices(Specification<Service> spec, Pageable pageable);

	ResponseEntity<ServicePageDTO> getServices(ServiceSearchDTO serviceSearchDTO);
	
	List<ServiceDTO> convertServicesToServiceDTOs(List<Service> services, ServiceConvertCriteriaDTO convertCriteria);

	ServiceDTO getServiceDTOById(Integer serviceId);







}

File: skyspaceeleven_ui/src/services/ServiceService.js
import http from "../http-common"; 

class ServiceService {
  getAllServices(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/service/services`, searchDTO);
  }

  get(serviceId) {
    return this.getRequest(`/service/${serviceId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/service?field=${matchData}`, null);
  }

  addService(data) {
    return http.post("/service/addService", data);
  }

  update(data) {
  	return http.post("/service/updateService", data);
  }
  
  uploadImage(data,serviceId) {
  	return http.postForm("/service/uploadImage/"+serviceId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ServiceService();
